### PR TITLE
sass-integration

### PR DESCRIPTION
**Pull request by Bartekus**
sass-loader & node-sass modules added / webpack.config.js modified to enable .scss extensions / linted ok

### CODE REVIEW

**Comment by Bartekus** [comment]
I figured that there is no harm in including sass plus I'm sure a lot of people will appreciate this down the line.

File: .npmignore [hunk 1]
```diff
@@ -24,7 +24,10 @@ build/Release
...
 # Dependency directory
 # https://www.npmjs.org/doc/misc/npm-faq.html#should-i-check-my-node_modules-folder-into-git
-node_modules
+node_modules/
 examples/biz-website/public/
 examples/blog/public/
 *.un~
+
+# IDEA/WebStorm configuration directory
+.idea/
```
Comment (KyleAMathews) [inline .npmignore]: Shouldn't this be added to the .gitignore?

File: lib/utils/webpack.config.js [hunk 2]
```diff
@@ -112,7 +112,7 @@ module.exports = (program, directory, stage, webpackPort = 1500, routes = []) =>
 
   function resolve () {
     return {
-      extensions: ['', '.js', '.jsx', '.cjsx', '.coffee', '.json', '.less', '.toml', '.yaml'],
+      extensions: ['', '.js', '.jsx', '.cjsx', '.coffee', '.json', '.less', '.scss', '.toml', '.yaml'],
```
Comment (KyleAMathews) [inline lib/utils/webpack.config.js]: Shouldn't sass be added here too?

**Comment by KyleAMathews** [comment]
:+1: yeah, I agree popular Webpack loaders should be added to Gatsby so things just *work* out of the box for the vast majority of people. I.e. add more from here https://www.npmjs.com/browse/keyword/webpack-loader

**Comment by Bartekus** [comment]
You are correct, along with .scss, there should also be .sass . I'll correct it and resubmit shortly.

File: .npmignore (hunk 1 above)
Comment (Bartekus) [inline .npmignore]: I'm not sure, I usually don't include it as not to pollute the project with unnecessary IDE configuration files

**Comment by Bartekus** [comment]
At this point you can merge this if you like. 
I'll open another ticket and start working on adding img loader.

File: .npmignore (hunk 1 above)
Comment (KyleAMathews) [inline .npmignore]: This is the npm ignore file not .gitignore.

**Comment by KyleAMathews** [comment]
Why is there 3 configs?

File: .npmignore (hunk 1 above)
Comment (Bartekus) [inline .npmignore]: ohh good catch, I feel silly, let me quickly fix it.

**Comment by Bartekus** [comment]
3 configs? I'm not sure I understand what you mean, but that's how I've always done it.

**Comment by KyleAMathews** [comment]
Oh... nvm :) I forgot that there's a config for each stage. Didn't look at the file in full.

File: .npmignore [hunk 3]
```diff
@@ -1,30 +0,0 @@
-# Logs
```
Comment (KyleAMathews) [inline .npmignore]: Could you restore this file.

**Comment by KyleAMathews** [comment]
I think once the `.npmignore` file is restored this is ready to go! Also if you could squish the commits down into one for cleanliness that'd be great.

**Comment by Bartekus** [comment]
I've tried to squash my commits but I'm not entirely convinced that I've been successful at it. ;(

**Comment by KyleAMathews** [comment]
:)

Try following this tutorial: https://github.com/ginatrapani/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit

**Comment by Bartekus** [comment]
I've re-submitted cleaned-up version on #110

**Pull request closed without merging**